docs/advanced/analytics/plugins.mdx

```diff
@@ -0,0 +1,340 @@
+---
+sidebar_position: 2
+title: Plugins
+description:
+  The analytics library is designed to work with any third party analytics tools
+  via plugins.
+---
+
+import AnalyticsFigure from "./assets/analytics.svg";
+import Figure from "@site/src/components/Figure";
+import ContactLink from "@site/src/components/ContactLink";
+import { useState } from "react";
+
+> _Since version 2.19_
+
+<p>{frontMatter.description}</p>
+
+Plugins are a powerful abstraction that let you:
+
+- add a new analytics provider (like google analytics or pixel)
+- hook into an existing analytics provider plugin
+- or add any kind of logic to react to visitor actions
+
+Plugins can be broken down into 2 types:
+
+- **Provider plugins** - connecting to third party analytic services
+- **Custom plugins** - additional features, data manipulation, & any other side
+  effects.
+
+Both have the same signature, as illustrated below.
+
```

Comment:
**nitpick:** it isn't super clear that these plugins are _just_  analytics plugins

**suggestion:** link to https://getanalytics.io/plugins so users can find the canonical documentation for analytics plugins and maybe also to https://getanalytics.io/plugins/writing-plugins/ from the same block (or in the section below?)